### PR TITLE
Feat/arm 310/radix switch input

### PR DESCRIPTION
**Pull request by tomRocketmakers**
## What's new?

- Created Radix Switch component

## Ticket number(s) in JIRA (if internal)

ARM-310

[board](https://rocketmakers.atlassian.net/jira/software/projects/HM/boards/172)

## Checklist

- [x] are your changes in Storybook?
- [x] are any breaking changes documented in `docs/migrating_from_oldstrong.md`?
- [x] does _everything_ have jsdoc?
- [x] is everything exported from index.ts?
- [x] are all new hooks added to `src/hooks/hooks.stories.mdx` or given their own docs in Storybook?
- [x] are all new SCSS mixins added to `src/theme/mixins.stories.mdx`?
- [x] are all new SCSS variables added to `src/theme/variables.stories.mdx`?

### CODE REVIEW

File: module-new/src/components/switch/switch.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  component: Switch,
+} as Meta<typeof Switch>;
+
+type Story = StoryObj<typeof Switch>;
+
+export const Default: Story = {
+  args: {
+    label: 'Airplane mode',
+  },
+  play: async ({ canvasElement }) => {
```
Comment (NathanRHD) [inline module-new/src/components/switch/switch.stories.tsx:19]: is it necessary to separately test that the form bindings (via the `bind` prop) are also hooked up correctly?

File: module-new/src/components/switch/switch.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  },
+};
+
+export const Disabled: Story = {
+  args: {
+    label: 'Switch is disabled',
+    disabled: true,
+    testId: 'switch-container',
+  },
+  play: async ({ canvasElement }) => {
```
Comment (NathanRHD) [inline module-new/src/components/switch/switch.stories.tsx:34]: perhaps testing that the value doesn't change on click when disabled?

File: module-new/src/components/switch/switch.theme.css [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  display: flex;
+  align-items: center;
+}
+
+.arm-switch-container[data-disabled='true'] {
+  filter: none;
+  opacity: 0.6;
+}
+
+.arm-switch-container[data-direction='vertical'] {
```
Comment (NathanRHD) [inline module-new/src/components/switch/switch.theme.css:13]: there is no vertical mode in the designs - if it just switching flex direction, is it necessary to have in armstrong vs just adding a custom class name in the consuming code?

Comment (tomRocketmakers) [reply module-new/src/components/switch/switch.theme.css:13]: Fair one - it was present in the previous version of the component so wasn't sure what to migrate. Happy to remove and do the same for the switch for readability/simplicity @jlews ?

Comment (jlews) [reply module-new/src/components/switch/switch.theme.css:13]: I vote for remove @tomRocketmakers in favour of simplicity, the consuming app can always override the classes for edge cases

File: module-new/src/components/switch/switch.component.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      required,
+      validationErrorMessages,
+      validationErrorsClassName,
+      scrollValidationErrorsIntoView,
+    },
+    ref
+  ) => {
+    const id = React.useId();
+
+    const globals = useArmstrongConfig({
```
Comment (jlews) [inline module-new/src/components/switch/switch.component.tsx:85]: You'll probably want to run a few more optional local props through globals, stuff like `validationErrorMode` and `validationIcon` etc. Take a look at `input.component.tsx` in module-new

**Review by jlews — CHANGES_REQUESTED** [review]

File: module-new/src/components/switch/switch.component.tsx (hunk 4 above)
Comment (jlews) [reply module-new/src/components/switch/switch.component.tsx:104]: It doesn't look like all the globals you've defined local overrides for are being used.

The rule with globals is, if you're control needs to use something that has a global setting, it should define a local prop for it, run it through the useArmstrongConfig hook, and then use the property in the control with `globals.x`, but if it's a global property that you don't need to read then you don't need to do any of this for that prop

Comment (tomRocketmakers) [reply module-new/src/components/switch/switch.component.tsx:104]: As the switch isn't using the Input wrapper and (to my knowledge) we're not using an error icon in this component, do we need all those props?

**Review by jlews — APPROVED** [review]

**Pull request merged**